pinot-common/src/main/java/org/apache/pinot/common/config/FieldConfig.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Properties;
+import javax.annotation.Nullable;
+
+
+public class FieldConfig extends BaseJsonConfig {
+  private String _name;
+  private EncodingType _encodingType;
+  private IndexType _indexType;
+  private Properties _properties;
```

Comment:
I don't think we need the functionalities provided by Properties class, maybe Map<String, String> is enough?

Reply:
Yes, Map<String, String> is enough. Properties is more for saving to file.